### PR TITLE
Pia 4596 rename feedback

### PR DESCRIPTION
**Pull request by zladzeyka**
- Deprecate old cleanup method
 - Rename sendFeedback - > sendTransferSummary
 - Extract cleanup () method
 - Update tests

### CODE REVIEW

File: BankAPILibrary/GiniBankAPILibraryExample/GiniBankAPILibraryExampleTests/TransferSummaryIntegrationTest.swift [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                                self.documentService.submitFeedback(for: compositeDocument, with: extractions) { result in
+                                    switch result {
+                                    case .success(_):
+
+                                        // 4. Verify that the extractions were updated
+                                        self.documentService.extractions(for: compositeDocument, cancellationToken: CancellationToken()) { result in
+                                            switch result {
+                                            case .success(let extractionResult):
+                                                let extractionsAfterFeedback = extractionResult.extractions
+                                                let fixtureExtractionsAfterFeedbackJson = self.loadFile(withName: "result_Gini_invoice_example_after_feedback", ofType: "json")
```
Comment (ValentinaIancu-Gini) [inline BankAPILibrary/GiniBankAPILibraryExample/GiniBankAPILibraryExampleTests/TransferSummaryIntegrationTest.swift:86]: [clarification]: do we want to rename the file accordingly from `result_Gini_invoice_example_after_feedback` to `result_Gini_invoice_example_after_transfer_summary`?
Also the properties: `extractionsAfterFeedback`, `fixtureExtractionsAfterFeedbackJson`, `fixtureExtractionsAfterFeedbackContainer`.

File: BankAPILibrary/GiniBankAPILibraryPinningExample/GiniBankAPILibraryPinningExampleTests/TransferSummaryIntegrationTest.swift [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                                self.documentService.submitFeedback(for: compositeDocument, with: extractions) { result in
+                                    switch result {
+                                    case .success(_):
+
+                                        // 4. Verify that the extractions were updated
+                                        self.documentService.extractions(for: compositeDocument, cancellationToken: CancellationToken()) { result in
+                                            switch result {
+                                            case .success(let extractionResult):
+                                                let extractionsAfterFeedback = extractionResult.extractions
+                                                let fixtureExtractionsAfterFeedbackJson = self.loadFile(withName: "result_Gini_invoice_example_after_feedback", ofType: "json")
```
Comment (ValentinaIancu-Gini) [inline BankAPILibrary/GiniBankAPILibraryPinningExample/GiniBankAPILibraryPinningExampleTests/TransferSummaryIntegrationTest.swift:104]: [clarification]: do we want to rename the file accordingly from `result_Gini_invoice_example_after_feedback` to `result_Gini_invoice_example_after_transfer_summary`?
Also the properties: `extractionsAfterFeedback`, `fixtureExtractionsAfterFeedbackJson`, `fixtureExtractionsAfterFeedbackContainer`.

File: BankSDK/GiniBankSDKExample/GiniBankSDKExample/Screen API/ScreenAPICoordinator.swift [hunk 3]
```diff
@@ -139,12 +139,13 @@ final class ScreenAPICoordinator: NSObject, Coordinator, UINavigationControllerD
...
 		}
 	
-		configuration.cleanup(paymentRecipient: extractedResults.first(where: { $0.name == "paymentRecipient"})?.value ?? "",
+		configuration.sendTransferSummary(paymentRecipient: extractedResults.first(where: { $0.name == "paymentRecipient"})?.value ?? "",
                               paymentReference: extractedResults.first(where: { $0.name == "paymentReference"})?.value ?? "",
                               paymentPurpose: extractedResults.first(where: { $0.name == "paymentPurpose"})?.value ?? "",
                               iban: extractedResults.first(where: { $0.name == "iban"})?.value ?? "",
                               bic: extractedResults.first(where: { $0.name == "bic"})?.value ?? "",
                               amountToPay: extractionAmount)
+        configuration.cleanup()
```
Comment (ValentinaIancu-Gini) [inline BankSDK/GiniBankSDKExample/GiniBankSDKExample/Screen API/ScreenAPICoordinator.swift:148]: [suggestion]: Indentation issue `configuration.sendTransferSummary`.
Please check also the indentation in this method `closeSreenAPIAndSendFeedback`.

File: BankSDK/GiniBankSDKExample/Tests/TransferSummaryIntegrationTest.swift [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+                DispatchQueue.main.asyncAfter(deadline: .now() + 10, execute: {
+                    // 5. Verify that the extractions were updated
+                    self.integrationTest.getUpdatedExtractionsFromGiniBankSDK(for: result.document!) { result in
+                        switch result {
+                        case let .success(extractionResult):
+                            let extractionsAfterFeedback = extractionResult.extractions
+
+
+                            let fixtureExtractionsAfterFeedbackJson = self.integrationTest.loadFile(withName: "result_Gini_invoice_example_payment_reference_after_feedback", ofType: "json")
```
Comment (ValentinaIancu-Gini) [inline BankSDK/GiniBankSDKExample/Tests/TransferSummaryIntegrationTest.swift:98]: [clarification]: do we want to rename the file accordingly from `result_Gini_invoice_example_payment_reference_after_feedback` to `result_Gini_invoice_example_payment_reference_after_transfer_summary`?
Also the properties: `extractionsAfterFeedback`, `fixtureExtractionsAfterFeedbackJson`, `fixtureExtractionsAfterFeedbackContainer`?

File: BankSDK/GiniBankSDKExample/Tests/TransferSummaryIntegrationTest.swift [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+                                                            candidates: [:])
+
+                        delegate.giniCaptureAnalysisDidFinishWith(result: analysisResult)
+
+                        GiniBankConfiguration.shared.lineItems = extractionResult.lineItems
+                        // 4. Send transfer summary for the extractions the user saw
+                        //    with the final (user confirmed or updated) extraction values
+                        GiniBankConfiguration.shared.sendTransferSummary(paymentRecipient: extractions["paymentRecipient"]?.value ?? "",
+                                                             paymentReference: extractions["paymentReference"]?.value ?? "",
+                                                             paymentPurpose: extractions["paymentPurpose"]?.value ?? "",
```
Comment (ValentinaIancu-Gini) [inline BankSDK/GiniBankSDKExample/Tests/TransferSummaryIntegrationTest.swift:188]: [siggestion]: Indentation issue with some of the method's parameters.

File: CaptureSDK/GiniCaptureSDKExample/Example Swift/ScreenAPICoordinator.swift [hunk 6]
```diff
@@ -109,12 +109,13 @@ extension ScreenAPICoordinator: UINavigationControllerDelegate {
 				extractionAmount = ExtractionAmount(value: Decimal(string: String(amountValue.split(separator: ":")[0])) ?? 0.0, currency: .EUR)
 			}
 		
-			visionConfiguration.cleanup(paymentRecipient: extractedResults.first(where: { $0.name == "paymentRecipient"})?.value ?? "",
+			visionConfiguration.sendTransferSummary(paymentRecipient: extractedResults.first(where: { $0.name == "paymentRecipient"})?.value ?? "",
```
Comment (ValentinaIancu-Gini) [inline CaptureSDK/GiniCaptureSDKExample/Example Swift/ScreenAPICoordinator.swift:112]: [suggestion]: Indentation issue `visionConfiguration.sendTransferSummary`.
Please check also the indentation in the entire method.

File: BankSDK/GiniBankSDKPinningExample/Tests/TransferSummaryIntegrationTest.swift [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+         result.extractions["amountToPay"]?.value = "950.00:EUR"
+
+         if result.extractions["amountToPay"] != nil {
+            DispatchQueue.main.asyncAfter(deadline: .now() + 10, execute: {
+               // 5. Verify that the extractions were updated
+               self.integrationTest.getUpdatedExtractionsFromGiniBankSDK(for: result.document!) { result in
+                  switch result {
+                  case let .success(extractionResult):
+                     let extractionsAfterFeedback = extractionResult.extractions
+                     let fixtureExtractionsAfterFeedbackJson = self.integrationTest.loadFile(withName: "result_Gini_invoice_example_after_feedback", ofType: "json")
```
Comment (ValentinaIancu-Gini) [inline BankSDK/GiniBankSDKPinningExample/Tests/TransferSummaryIntegrationTest.swift:119]: [clarification]: do we want to rename the file accordingly from `result_Gini_invoice_example_after_feedback` to `result_Gini_invoice_example_after_transfer_summary`?
Also the properties: `extractionsAfterFeedback, fixtureExtractionsAfterFeedbackJson, fixtureExtractionsAfterFeedbackContainer`.

File: BankSDK/GiniBankSDKPinningExample/Tests/TransferSummaryIntegrationTest.swift [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+                                                      lineItems: extractionResult.lineItems,
+                                                      images: [],
+                                                      document: self.giniCaptureSDKDocumentService?.document, candidates: [:])
+
+                  delegate.giniCaptureAnalysisDidFinishWith(result: analysisResult)
+                  // 4. Send transfer summary for the extractions the user saw
+                  //    with the final (user confirmed or updated) extraction values
+                  GiniBankConfiguration.shared.sendTransferSummary(paymentRecipient: extractions["paymentRecipient"]?.value ?? "",
+                                                       paymentReference: extractions["paymentReference"]?.value ?? "",
+                                                       paymentPurpose: extractions["paymentPurpose"]?.value ?? "",
```
Comment (ValentinaIancu-Gini) [inline BankSDK/GiniBankSDKPinningExample/Tests/TransferSummaryIntegrationTest.swift:188]: [suggestion]: Please fix indentation.

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/GiniConfiguration.swift [hunk 9]
```diff
@@ -528,4 +531,84 @@ import GiniBankAPILibrary
...
+     - Send the transfer summary after TAN verification and provide the extraction values the user has used.
+
+     - parameter paymentRecipient: paymentRecipient description
+     - parameter paymentReference: paymentReference description
+     - parameter iban: iban description
+     - parameter bic: bic description
+     - parameter amountToPay: amountToPay description
+     */
+    public func sendTransferSummary(paymentRecipient: String,
+                                   paymentReference: String,
```
Comment (ValentinaIancu-Gini) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/GiniConfiguration.swift:555]: [suggestion]: Please indentation of parameters.

File: CaptureSDK/GiniCaptureSDKExample/Tests/TransferSummaryIntegrationTest.swift [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+
+            if result.extractions["amountToPay"] != nil {
+                DispatchQueue.main.asyncAfter(deadline: .now() + 10, execute: {
+                    // 5. Verify that the extractions were updated
+
+                    self.integrationTest.getUpdatedExtractionsFromGiniCaptureSDK(for: result.document!) { result in
+                        switch result {
+                        case let .success(extractionResult):
+                            let extractionsAfterFeedback = extractionResult.extractions
+                            let fixtureExtractionsAfterFeedbackJson = self.integrationTest.loadFile(withName: "result_Gini_invoice_example_after_feedback", ofType: "json")
```
Comment (ValentinaIancu-Gini) [inline CaptureSDK/GiniCaptureSDKExample/Tests/TransferSummaryIntegrationTest.swift:99]: [clarification]: do we want to rename the file accordingly from `result_Gini_invoice_example_after_feedback` to `result_Gini_invoice_example_after_transfer_summary`?
Also the properties: `extractionsAfterFeedback, fixtureExtractionsAfterFeedbackJson, fixtureExtractionsAfterFeedbackContainer`.

File: CaptureSDK/GiniCaptureSDKExample/Tests/TransferSummaryIntegrationTest.swift [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+                                                            lineItems: extractionResult.lineItems,
+                                                            images: [],
+                                                            document: self.giniCaptureSDKDocumentService?.document, candidates: [:])
+
+                        delegate.giniCaptureAnalysisDidFinishWith(result: analysisResult)
+                        // 4. Send transfer summary for the extractions the user saw
+                        //    with the final (user confirmed or updated) extraction values
+                        GiniConfiguration.shared.sendTransferSummary(paymentRecipient: extractions["paymentRecipient"]?.value ?? "",
+                                                         paymentReference: extractions["paymentReference"]?.value ?? "",
+                                                         paymentPurpose: extractions["paymentPurpose"]?.value ?? "",
```
Comment (ValentinaIancu-Gini) [inline CaptureSDK/GiniCaptureSDKExample/Tests/TransferSummaryIntegrationTest.swift:170]: [suggestion]: Please fix the indentation of the parameters.

File: CaptureSDK/GiniCaptureSDKPinningExample/Tests/TransferSummaryIntegrationTest.swift [hunk 12]
```diff
@@ -0,0 +1,219 @@
...
+            result.extractions["amountToPay"]?.value = "950.00:EUR"
+
+            if result.extractions["amountToPay"] != nil {
+                DispatchQueue.main.asyncAfter(deadline: .now() + 10, execute: {
+                    // 5. Verify that the extractions were updated
+                    self.integrationTest.getUpdatedExtractionsFromGiniCaptureSDK(for: result.document!) { result in
+                        switch result {
+                        case let .success(extractionResult):
+                            let extractionsAfterFeedback = extractionResult.extractions
+                            let fixtureExtractionsAfterFeedbackJson = self.integrationTest.loadFile(withName: "result_Gini_invoice_example_after_feedback", ofType: "json")
```
Comment (ValentinaIancu-Gini) [inline CaptureSDK/GiniCaptureSDKPinningExample/Tests/TransferSummaryIntegrationTest.swift:116]: [clarification]: do we want to rename the file accordingly from `result_Gini_invoice_example_after_feedback` to `result_Gini_invoice_example_after_transfer_summary`?
Also the properties: `extractionsAfterFeedback, fixtureExtractionsAfterFeedbackJson, fixtureExtractionsAfterFeedbackContainer`.

File: CaptureSDK/GiniCaptureSDKPinningExample/Tests/TransferSummaryIntegrationTest.swift [hunk 13]
```diff
@@ -0,0 +1,219 @@
...
+                        let analysisResult = AnalysisResult(extractions: extractions,
+                                                            lineItems: extractionResult.lineItems,
+                                                            images: [],
+                                                            document: self.giniCaptureSDKDocumentService?.document, candidates: [:])
+
+                        delegate.giniCaptureAnalysisDidFinishWith(result: analysisResult)
+                        // 4. Send transfer summary for the extractions the user saw
+                        //    with the final (user confirmed or updated) extraction values
+                        GiniConfiguration.shared.sendTransferSummary(paymentRecipient: extractions["paymentRecipient"]?.value ?? "",
+                                                         paymentReference: extractions["paymentReference"]?.value ?? "",
```
Comment (ValentinaIancu-Gini) [inline CaptureSDK/GiniCaptureSDKPinningExample/Tests/TransferSummaryIntegrationTest.swift:186]: [suggestion]: Please fix the indentation of the parameters.

File: BankAPILibrary/GiniBankAPILibraryExample/GiniBankAPILibraryExampleTests/TransferSummaryIntegrationTest.swift (hunk 1 above)
Comment (zladzeyka) [reply BankAPILibrary/GiniBankAPILibraryExample/GiniBankAPILibraryExampleTests/TransferSummaryIntegrationTest.swift:86]: I will keep the files names for now

File: BankAPILibrary/GiniBankAPILibraryPinningExample/GiniBankAPILibraryPinningExampleTests/TransferSummaryIntegrationTest.swift (hunk 2 above)
Comment (zladzeyka) [reply BankAPILibrary/GiniBankAPILibraryPinningExample/GiniBankAPILibraryPinningExampleTests/TransferSummaryIntegrationTest.swift:104]: We keep the file names

File: BankSDK/GiniBankSDKExample/Tests/TransferSummaryIntegrationTest.swift (hunk 4 above)
Comment (zladzeyka) [reply BankSDK/GiniBankSDKExample/Tests/TransferSummaryIntegrationTest.swift:98]: We keep the current names

File: BankSDK/GiniBankSDKPinningExample/Tests/TransferSummaryIntegrationTest.swift (hunk 7 above)
Comment (zladzeyka) [reply BankSDK/GiniBankSDKPinningExample/Tests/TransferSummaryIntegrationTest.swift:119]: We keep the file names

File: CaptureSDK/GiniCaptureSDKExample/Tests/TransferSummaryIntegrationTest.swift (hunk 10 above)
Comment (zladzeyka) [reply CaptureSDK/GiniCaptureSDKExample/Tests/TransferSummaryIntegrationTest.swift:99]: we keep the names for now

File: CaptureSDK/GiniCaptureSDKPinningExample/Tests/TransferSummaryIntegrationTest.swift (hunk 12 above)
Comment (zladzeyka) [reply CaptureSDK/GiniCaptureSDKPinningExample/Tests/TransferSummaryIntegrationTest.swift:116]: we keep the names for now

**Review by ValentinaIancu-Gini — APPROVED** [review]

**Pull request merged**